### ISSUE TITLE
Proper Link Previews

### ISSUE THREAD

**Issue by camball** [camball/camball.io#36]
## Motivation

Currently, link previews for the blog look like the following:

![Image](https://github.com/user-attachments/assets/d1bd300b-8cb2-407c-b207-f37a24695bcb)

The first is for the home page, then the second is for the article "Canon Software Engineer Events". The first is acceptable. The second sucks and absolutely must be fixed. It doesn't even include the blog article title.

## How?

- [Open Graph Protocol Specification](https://ogp.me)
- [Blog article on creating link previews](https://andrejgajdos.com/how-to-create-a-link-preview/)
- [How Slack creates link previews](https://medium.com/slack-developer-blog/everything-you-ever-wanted-to-know-about-unfurling-but-were-afraid-to-ask-or-how-to-make-your-e64b4bb9254)
- [Open Graph Generator – Article](https://webcode.tools/open-graph-generator/article)
- Twitter/X
  - [Card Meta Tags](https://developer.x.com/en/docs/x-for-websites/cards/overview/summary)
  - Test with [Card Validator](https://cards-dev.x.com/validator)

**Comment by camball** [comment]
Fixed the iMessage one completely—it looks great:

![Screenshot 2024-09-15 at 5 16 24 AM](https://github.com/user-attachments/assets/4819a4a0-c282-49fb-877f-9d0d8e368e3b)

Obsidian one is better, but leaves much to be desired:

![Screenshot 2024-09-15 at 5 16 49 AM](https://github.com/user-attachments/assets/98b18f4b-1c1f-414b-b7fa-6e9f4d21351e)

For now, this is complete; I'm happy with the results.

**Comment by camball** [comment]
Passes validation in the [Twitter Card Validator](https://cards-dev.x.com/validator):

```txt
INFO:  Page fetched successfully
INFO:  16 metatags were found
INFO:  twitter:card = summary tag found
INFO:  Card loaded successfully
```

**Comment by camball** [comment]
Update: it's not working for Slack, which is ironic because I followed Slack's engineering blog to implement it. Need to fix this ASAP.

**Comment by camball** [comment]
Update: only partially works for Twitter. Gets title and generates *a* preview, but missing the image:

![image](https://github.com/user-attachments/assets/f1826984-162e-4640-a41a-b05302b3af96)

**Comment by camball** [comment]
Update: with 0815d28, link previews work on Slack. I did have a hunch the reason links wouldn't display in a pretty manner was because of the missing description, and it turns out that was correct after testing.

![successful-slack-link-unfurl](https://github.com/user-attachments/assets/0ff80586-a231-48fd-80d9-fc33cccf866b)

It's still missing the favicon though, although I didn't even know I had to do a special notation for the favicons, so I'll need to investigate. Medium articles have everything set for a proper unfurl, so use any good Medium article as a reference when fixing.

Twitter is still broken after 0815d28; neither an image nor the new descriptions come up.
